File: Taller Tipos de clases/JAVA/Clase Abstracta/Figura.java
abstract class Figura {

     String color;

    // Constructor sin parámetros
    public Figura() {
        color = "Negro";
    }

    // Constructor con parámetros
    public Figura(String color) {
        this.color = color;
    }

    // Constructor copia
    public Figura(Figura f) {
        this.color = f.color;
    }

    // Método abstracto
    abstract double area();
}

File: Taller Tipos de clases/Go/Clase Abstracta/Figura.go
abstract class Figura
{
    public string Color;

    public Figura()
    {
        Color = "Negro";
    }

    public Figura(string color)
    {
        Color = color;
    }

    public Figura(Figura f)
    {
        Color = f.Color;
    }

    public abstract double Area();
}

